browser/plugins/annotation.plugin.js

```diff
@@ -8,36 +14,38 @@ var Annotation = E2.plugins.annotation = function() {
 	
 	this.output_slots = []
 	
-	this.state = { text: '', width: 0, height: 0 }
+	this.state = { text: '', width: minWidth, height: minHeight }
 }
 
 Annotation.prototype = Object.create(Plugin.prototype)
 
 Annotation.prototype.create_ui = function() {
 	var that = this
-	var inp = $('<textarea placeholder="Type text here" />')
-	inp.css({
-		'resize': true
-	})
-	
-	inp.on('change', function() {
-		that.undoableSetState('text', inp.val(), that.state.text)
-	})
-	
-	// Chrome doesn't handle resize properly for anything but the window object,
-	// so we store the potentially altered size of the textarea on mouseup.
-	inp.mouseup(function() {
-		var ta = $(this)
 
-		that.state.width = ta.width()
-		that.state.height = ta.height()
+	var adapter = {
+		get text()  	{ return that.state.text; },
+		set text(str) 	{ return that.state.text = str},
+		get width()  	{ return that.state.width },
+		set width(px) 	{ return that.state.width = clamp(px, minWidth, maxWidth) },
+		get height()  	{ return that.state.height },
+		set height(px) 	{ return that.state.height = clamp(px, minHeight, maxHeight) }
+	}
+
+	var onChange 		= this.updateUi.bind(this)
+	var onBeginChange 	= this.beginBatchModifyState.bind(this)
+	var onEndChange 	= function() { that.endBatchModifyState('Annotation') }
```

Comment:
i was hoping to handle that in a separate PR. similar UI issues are building up.